backend/templates/create-report/template-files/routes/index.js

```diff
@@ -1,21 +1,18 @@
-const controller = require("@controllers/reports/$reportName.js");
+const getControllerFrom = require("@controllers/reports/genericReportController");
 // const validators = require("@validators/report");
-const what = "report";
 
-const routes = [
-  {
-    method: "GET",
-    url: `/${what}/projects/$reportName`,
-    onRequest: controller.$reportName,
-    handler: controller.getReport,
-  },
-];
-const registerRoutes = (fastify, options, done) => {
-  // Ensure all of the routes above get registered.
-  routes.forEach((route) => fastify.route(route));
-  done();
-};
+const name = `$reportName`;
+const controller = getControllerFrom(name);
 
 module.exports = {
-  registerRoutes,
+  registerRoutes: (fastify, options, done) => {
+    fastify.route({
+      method: "GET",
+      url: `/report/projects/${name}`,
+      // schema: getAll,
```

Comment:
if you have this validator working, we might want to remove the comments.

Reply:
the trouble with validators is they handle input AND output, so I think we'd need to customize the validator for each new report. 

I'm open to suggestions if you want to look at making a generic validator though. I think it would be large and complex, but it could save space.